pkg/driver/type/modernbpf.go

```diff
@@ -51,6 +56,34 @@ func (m *modernBpf) HasArtifacts() bool {
 	return false
 }
 
+// Get the private symbol `probeProgram` that will be used to test for
+// type Tracing, attachType AttachTraceRawTp program availability.
+//
+//go:linkname probeProgram github.com/cilium/ebpf/features.probeProgram
+func probeProgram(spec *ebpf.ProgramSpec) error
```

Comment:
To avoid fully rewriting the function, we "unsafely" import it using `go:linkname`.
I think this is better than copying the function locally, because then we get any change `probeProgram` might get with new versions, and if `probeProgram` changes signature/name, our build will fail while bumping the library dep.
This way, we always know if something has changed.